### PR TITLE
[Chore] TypeScript lint, test and type warnings and errors

### PR DESCRIPTION
**Pull request by beau-charman**
This PR cleans up and addresses a bunch of Typescript lint, test and type warnings and errors

- Some no return types
- Some prop / object types
- Unused variables 
- Properties on objects without being types
- Non- existent CSS class references 
- Test for ProductDetailHeader (which surfaced a bunch of errors)
- Migration of some FC to CWC (ComponentWith/outChildren)

### CODE REVIEW

File: src/components/Select/Select.fixture.ts [hunk 1]
```diff
@@ -4,9 +4,9 @@ const SelectFixture = {
   errorMessage: 'Test Error Messsage',
   label: 'Vestibulum accumsan',
   name: 'select',
-  onBlur: () => undefined,
-  onChange: () => undefined,
-  onFocus: () => undefined,
+  onBlur: (): undefined => undefined,
+  onChange: (): undefined => undefined,
+  onFocus: (): undefined => undefined,
```
Comment (MkMan) [inline src/components/Select/Select.fixture.ts:9]: I thought TS would be smart enough to deduce this 🤔

**Review by MkMan — APPROVED** [review]
Good stuff, @beau-charman 👏 Thanks

external-content duckduckgo com

File: src/components/Heading/Heading.types.ts [hunk 2]
```diff
@@ -16,4 +16,4 @@ type HeadingProps = {
   theme?: Themes;
 };
 
-export type { HeadingProps };
+export type { HeadingProps, Levels, Sizes };
```
[the remark this answers is not in the document]
Comment (beau-charman) [reply src/components/Heading/Heading.types.ts:19]: This time I will change it. Usually the `components/Heading/Heading.types` directory would be enough to distinguish which `Levels` you would be importing.

File: src/components/Select/Select.fixture.ts (hunk 1 above)
Comment (beau-charman) [reply src/components/Select/Select.fixture.ts:9]: Well it was throwing errors about being untyped. But know its not :( my linters and prettier are just not trustworthy anymore

**Pull request merged**